shared/src/uniswap_pool.rs

```diff
@@ -69,25 +55,29 @@ impl Pool {
     /// Given an output amount and token, returns a required input amount and address of the other asset.
     /// Returns None if operation not possible due to arithmetic issues (e.g. over or underflow, reserve too small)
     pub fn get_amount_in(&self, token_out: H160, amount_out: U256) -> Option<(U256, H160)> {
+        let (reserve_out, reserve_in, token_in) = self.get_reserves(token_out);
+        Some((
+            self.amount_in(amount_out, reserve_in, reserve_out)?,
+            token_in,
+        ))
+    }
+
+    fn get_reserves(&self, token: H160) -> (U256, U256, H160) {
         // https://github.com/Uniswap/uniswap-v2-periphery/blob/master/contracts/libraries/UniswapV2Library.sol#L53
```

Comment:
IMO with this signature it's hard to understand what the function is doing without seeing how it is called and I feel the code became harder to read despite eliminating some duplicated code. It's unclear what the parameter is and how it relates to the return type.
Maybe it can be refactored in a way that that is clear or we keep the duplicate code.

Reply:
I would argue the code was already hard to read and that this change doesn't make it any harder. I will aim to make this more clear by providing better description, names and function signature.

Reply:
I have gone and fully understood the purpose of the method, renamed it, defined a type and documented the functionality. I think it is much more clear now. Could probably cut a bit of the `get_relative_reserves` doc-string since the type is very well defined. Let me know what you think.

Reply:
Your doc comments makes it very clear, thanks.